### PR TITLE
Use make-fetch-happen instead of request

### PR DESCRIPTION
**Pull request by CamilleDrapier**
cf: https://github.com/sass/node-sass/issues/2851

Follows the implementation that was done here: https://github.com/sass/node-sass/pull/2961 ; also remove `npmlog` that seems unused from here on.

Because the latest node-gyp does not support python2 (https://github.com/nodejs/node-gyp/blob/master/CHANGELOG.md#810-2021-05-28), use python3 for all nodes builds/tests. (Not sure if `src/libsass/script/ci-report-coverage:8` should be affected?)

### CODE REVIEW

**Comment by CamilleDrapier** [comment]
Sorry for taking your time with this, I was merely trying to see if this would be reasonable enough.. while doing the changes locally it seemed okayish~~ but after creating the PR, I thought it might involve too many things that I did not understand enough.

Let's try to only replace the request/fetch, I missed seeing that the [strictSSL/rejectUnauthorized](https://github.com/zkat/make-fetch-happen#--optsca-optscert-optskey-optsstrictssl) option was supported by `make-fetch-happen`, but this rename probably impacts the PR you mentioned :bow:

File: test/downloadoptions.js [hunk 1]
```diff
@@ -59,31 +57,11 @@ describe('util', function() {
...
-            'User-Agent': ua(),
-          },
-          encoding: null,
-        };
-
-        assert.deepStrictEqual(opts(), expected);
-      });
-    });
-
-    describe('with SASS_REJECT_UNAUTHORIZED set to false', function() {
```
Comment (xzyfer) [inline test/downloadoptions.js:74]: If we restore this test I think we're ok to proceed.

**Comment by xzyfer** [comment]
I've rebase this on the latest master

**Review by xzyfer — APPROVED** [review]
This seems reasonable to me. If @nschonni has no reservations I'm happy to roll give it a shot.

**Review by nschonni — APPROVED** [review]
I'm sure will get some weird edge cases not matter what library we use, so I'm OK with giving this a try 😄

**Comment by xzyfer** [comment]
Thanks for your work @CamilleDrapier

**Comment by xzyfer** [comment]
This was accidentally released in v7.0.2 and subsequently broke binary downloads. See #3297

**Comment by CamilleDrapier** [comment]
Sorry about this! Not really sure what is the underlying problem, but please let me know if I can help :bow:

**Comment by xzyfer** [comment]
No worries @CamilleDrapier, it was my mistake. My comment was intended for posterity. I've opened https://github.com/sass/node-sass/issues/3300 to track fixing it before releasing 8.0.0.

**Pull request merged**